<launch>

  <!-- usb_cam -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" >
    <param name="video_device"     value="/dev/videoCAM"                                />
    <param name="autoexposure"     value="false"                                        />
    <param name="exposure"         value="60"                                           />
    <param name="image_width"      value="640"                                          />
    <param name="image_height"     value="480"                                          />
    <param name="pixel_format"     value="yuyv"                                         />
    <param name="camera_frame_id"  value="usb_cam"                                      />
    <param name="camera_name"      value="usb_cam"                                      />
    <param name="camera_info_url"  value="file://$(find usb_cam)/calibration/cal.yaml"  />
    <param name="io_method"        value="mmap"                                         />
  </node>


  <!-- ar tag -->
  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" >
    <param name="marker_size"                type="double" value="9.0" />
    <param name="max_new_marker_error"       type="double" value="0.05" />
    <param name="max_track_error"            type="double" value="0.05" />
    <param name="output_frame"               type="string" value="usb_cam" />
    <remap from="camera_image" to="/usb_cam/image_raw"/>
    <remap from="camera_info" to="/usb_cam/camera_info"/>
  </node>


  <!-- ultrasonic -->
  <node pkg="xycar_ultrasonic" type="xycar_ultrasonic.py" name="xycar_ultrasonic" />

  
  <!-- 9dof IMU -->
  <!-- <node pkg="xycar_imu" type="9dof_imu_node.py" name="xycar_imu" > -->
  <!--  <param name="rviz_mode" type="string" value="true" /> -->
  <!-- </node> -->
  

  <!-- xycar_b2 motor -->
  <node name="xycar_motor" pkg="xycar_motor" type="xycar_motor_b2.py" >
    <param name="angle_offset" value="0" />
  </node>

  
  <!-- darknet_ros -->
  <rosparam command="load" ns="darknet_ros" file="$(find darknet_ros)/config/ros.yaml"/>
  <rosparam command="load" ns="darknet_ros" file="$(find darknet_ros)/config/yolov2.yaml"/>

  <node pkg="darknet_ros" type="darknet_ros" name="darknet_ros" output="screen" launch-prefix="">
  <param name="weights_path"          value="$(find darknet_ros)/yolo_network_config/weights" />
  <param name="config_path"           value="$(find darknet_ros)/yolo_network_config/cfg" />
  </node>

  

  <!-- main -->
  <node name="rally" pkg="rally" type="main.py" output="screen" />

</launch>
